internal/evidence/pool.go

```diff
@@ -316,6 +333,19 @@ func (evpool *Pool) addPendingEvidence(ev types.Evidence) error {
 	}
 
 	atomic.AddUint32(&evpool.evidenceSize, 1)
+	evpool.Metrics.NumEvidence.Set(float64(evpool.evidenceSize))
+
+	// This should normally never be false
```

Comment:
Is it worth logging, in that case? Or maybe report an error? It would indeed be kind of weird to not have an event bus.

Reply:
The only case where this happens right now is in tests. As I am forwarding the eventbus from node/setup.go when running tendermint we should in theory always have an eventbus. I saw somewhere that there was a panic thrown in case the eventbus is nil but I am not 100% sure whether its worth aborting the program due to this. I would actually add a check for whether the eventbus is running and if not explicitly set this to nil. This should probably be done in OnStart of the evidence reactor and then we explicitly set this to nil if it is not. This way we'd avoid the situation I had with the tests, where things block. Again - this should not happen, but maybe down the road it might. The reason I would add this check in OnStart is that the function  (eventBus.isRunning() )  has a mutex and it would be costly to do it whenever an event is published.

Reply:
I agree panicking is not the right approach here. I was more thinking we might simply return an error. Even in a test I think we'd want to explicitly set the bus rather than silently falling back on the lazy init case?

Reply:
> Ok..in the tests however we have to start the eventbus as well in order for them not to block even though they do not really touch on that part of the code. 

Yeah, the way these various components all have to be "started" explicitly is kind of frictional. I agree, though, that it's probably better to emulate the "normal" flow even though it isn't necessary, than to special-case the test.

Reply:
Fixed by replacing error type returned.